flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -530,8 +539,10 @@ public void startTaskThread() {
 	@Override
 	public void run() {
 		try {
+			MDC.put("jobName", this.jobName);
 			doRun();
 		} finally {
+			MDC.remove("jobName");
 			terminationFuture.complete(executionState);
```

Comment:
Don't we want to have the same changes in `LegacySourceFunctionThread` as well?

Reply:
Depends a bit. From what I saw in the log4j code, by default the thread context is not inherited to child threads, so we would actually need to setup the MDC whenever any thread is created, which is naturally terrible.
If the `isThreadContextMapInheritable` property is set to `true`, then all threads created within a thread inherit the MDC (as a copy), and then we shouldn't need to do anything.